pkg/ingester/ingester_v2.go

```diff
@@ -1483,6 +1497,10 @@ func (i *Ingester) v2QueryStreamChunks(ctx context.Context, db *userTSDB, from,
 func (i *Ingester) getTSDB(userID string) *userTSDB {
 	i.userStatesMtx.RLock()
 	defer i.userStatesMtx.RUnlock()
+	if i.TSDBState.closed {
```

Comment:
I don't think we should check the flag here. We should instead check for closing state in query functions (v2LabelNames, v2LabelValues, ...). Behaviour of query methods will be more explicit that way.

Reply:
I've removed the check from this function and added to the query functions.